engine/collection/compliance/core.go

```diff
@@ -191,36 +230,37 @@ func (c *Core) OnBlockProposal(originID flow.Identifier, proposal *messages.Clus
 }
 
 // processBlockAndDescendants is a recursive function that processes a block and
-// its pending proposals for its children. By induction, any children connected
-// to a valid proposal are validly connected to the finalized state and can be
+// its pending descendants. By induction, any child block of a
+// valid proposal is itself connected to the finalized state and can be
 // processed as well.
-func (c *Core) processBlockAndDescendants(proposal *messages.ClusterBlockProposal) error {
+// No errors are expected during normal operation.
+func (c *Core) processBlockAndDescendants(proposal *messages.ClusterBlockProposal, parent *flow.Header) error {
 	blockID := proposal.Header.ID()
+	log := c.log.With().
+		Str("block_id", blockID.String()).
+		Uint64("block_height", proposal.Header.Height).
+		Uint64("block_view", proposal.Header.View).
+		Uint64("parent_view", parent.View).
+		Logger()
 
 	// process block itself
-	err := c.processBlockProposal(proposal)
-	// child is outdated by the time we started processing it
-	// => node was probably behind and is catching up. Log as warning
-	if engine.IsOutdatedInputError(err) {
-		c.log.Info().Msg("dropped processing of abandoned fork; this might be an indicator that the node is slightly behind")
-		return nil
-	}
-	// the block is invalid; log as error as we desire honest participation
-	// ToDo: potential slashing
-	if engine.IsInvalidInputError(err) {
-		c.log.Warn().
-			Err(err).
-			Bool(logging.KeySuspicious, true).
-			Msg("received invalid block from other node (potential slashing evidence?)")
-		return nil
-	}
-	if engine.IsUnverifiableInputError(err) {
-		c.log.Warn().
-			Err(err).
-			Msg("received unverifiable from other node")
-		return nil
-	}
+	err := c.processBlockProposal(proposal, parent)
 	if err != nil {
+		if checkForAndLogOutdatedInputError(err, log) {
+			return nil
+		}
+		if checkForAndLogInvalidInputError(err, log) || checkForAndLogUnverifiableInputError(err, log) {
```

Comment:
⚠️ We should differentiate unverifiable and invalid inputs here. 
* unknown epoch no longer results in unverifiable input error
* cluster state now reports unverifiable blocks correctly (https://github.com/onflow/flow-go/pull/3577) -> in this case we should log a warning, but not mark the block invalid in VoteAggregator
```suggestion
		if checkForAndLogOutdatedInputError(err, log) || checkForAndLogUnverifiableInputError(err, log) {
			return nil
		}
		if checkForAndLogInvalidInputError(err, log) {
```